compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -81,47 +89,47 @@ impl Function {
-        let caller_runtime = self.runtime();
-
-        let should_inline_call =
-            |_context: &PerFunctionContext, ssa: &Ssa, called_func_id: FunctionId| -> bool {
-                // Do not inline self-recursive functions on the top level.
-                // Inlining a self-recursive function works when there is something to inline into
-                // by importing all the recursive blocks, but for the entry function there is no wrapper.
-                if called_func_id == self.id() {
-                    return false;
-                }
-                let callee = &ssa.functions[&called_func_id];
+    ) -> bool {
+        // Do not inline self-recursive functions on the top level.
+        // Inlining a self-recursive function works when there is something to inline into
+        // by importing all the recursive blocks, but for the entry function there is no wrapper.
+        if called_func_id == self.id() {
+            return false;
+        }
+        let callee = &ssa.functions[&called_func_id];
 
-                match callee.runtime() {
-                    RuntimeType::Acir(inline_type) => {
-                        // If the called function is acir, we inline if it's not an entry point
+        match callee.runtime() {
+            RuntimeType::Acir(inline_type) => {
+                // If the called function is acir, we inline if it's not an entry point
 
-                        // If we have not already finished the flattening pass, functions marked
-                        // to not have predicates should be preserved.
-                        let preserve_function =
-                            !inline_no_predicates_functions && callee.is_no_predicates();
+                // If we have not already finished the flattening pass, functions marked
+                // to not have predicates should be preserved.
+                let preserve_function =
+                    !inline_no_predicates_functions && callee.is_no_predicates();
 
-                        !inline_type.is_entry_point() && !preserve_function
-                    }
-                    RuntimeType::Brillig(_) => {
-                        if caller_runtime.is_acir() {
-                            // We never inline a brillig function into an ACIR function.
-                            return false;
-                        }
-                        // We inline inline if the function called wasn't ruled out as too costly or recursive.
-                        inline_infos
-                            .get(&called_func_id)
-                            .map(|info| info.should_inline)
-                            .unwrap_or_default()
-                    }
+                !inline_type.is_entry_point() && !preserve_function
+            }
+            RuntimeType::Brillig(_) => {
+                if self.runtime().is_acir() {
+                    // We never inline a brillig function into an ACIR function.
+                    return false;
```

Comment:
ditto here and for inlining ACIR into brillig.

Reply:
> and for inlining ACIR into brillig

Is that a new check that should be added?